artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/invm/InVMConnection.java

```diff
@@ -237,6 +240,22 @@ public void run() {
 
    }
 
+   @Override
+   public void write(ActiveMQBuffer buffer,
+                     RandomAccessFile raf,
+                     FileChannel fileChannel,
+                     long offset,
+                     int dataSize,
+                     boolean closeChannel) {
+      try {
+         if (closeChannel)
```

Comment:
This method is correct that's empty?
Anyway will use a proper logger here

Reply:
In ReplicationSyncFileMessage::encode, the file is read into buffer for non netty connection, so we just need to send buffer.